packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -217,20 +217,46 @@ describe('KeyringController', () => {
   });
 
   describe('createNewVaultAndKeychain', () => {
-    it('should create new vault, mnemonic and keychain', async () => {
-      const initialSeedWord = await keyringController.exportSeedPhrase(
-        password,
-      );
-      expect(initialSeedWord).toBeDefined();
-      const currentState = await keyringController.createNewVaultAndKeychain(
-        password,
-      );
-      expect(initialState).not.toBe(currentState);
-      const currentSeedWord = await keyringController.exportSeedPhrase(
-        password,
-      );
-      expect(currentSeedWord).toBeDefined();
-      expect(initialSeedWord).not.toBe(currentSeedWord);
+    describe('when there is no existing vault', () => {
+      it('should create new vault, mnemonic and keychain', async () => {
+        const cleanKeyringController = new KeyringController(
+          preferences,
+          baseConfig,
+        );
+        const initialSeedWord = await keyringController.exportSeedPhrase(
+          password,
+        );
+        const currentState =
```

Comment:
We are testing the return value of the method, which is good, but should we also test the state of the controller directly as well? Same goes for the next test. (Unless `exportSeedPhrase` returns a part of the state already?)

Reply:
The return value _is_ the controller state. At least it's the in-memory state, which is the more informative piece (the persisted state is just the encrypted vault).

Is there something else in particular you had in mind that we should be testing?

Reply:
If that's the case then I suppose it's not required. My thought was that if we ask the controller directly it makes this test more future-proof. But I suppose this would do.

Reply:
Making this access the state in a more conventional way sounds like a good idea to me too. But this is a bit of a weird case. The "real" controller state right now is just the persisted state; the in-memory state would need to be accessed via a custom property (not exactly future proof).